### PR TITLE
fix: use pagination from hubspot api

### PR DESCRIPTION
**Pull request by RonaldEAM**
Changes:

- Updated @hubspot/api-client
- Refactored manual requests to use new methods from hubspot api sdk
- Added pagination functionality to api requests that support it

### CODE REVIEW

**Review by VDubber — COMMENTED** [review]
A few items of feedback.

File: src/steps/owners/__recordings__/fetchOwners_1399607903/recording.har [hunk 1]
```diff
@@ -4,76 +4,265 @@
...
         "response": {
-          "bodySize": 404,
+          "bodySize": 207,
           "content": {
+            "encoding": "base64",
             "mimeType": "application/json;charset=utf-8",
-            "size": 404,
-            "text": "{\"results\":[{\"id\":\"84106262\",\"email\":\"stefan@creativice.com\",\"firstName\":\"Stefan\",\"lastName\":\"Virag\",\"userId\":25150127,\"createdAt\":\"2021-06-14T19:38:46.907Z\",\"updatedAt\":\"2021-06-14T19:38:46.907Z\",\"archived\":false},{\"id\":\"90089236\",\"email\":\"viragsf@gmail.com\",\"firstName\":\"\",\"lastName\":\"\",\"userId\":25516621,\"createdAt\":\"2021-07-13T13:57:46.581Z\",\"updatedAt\":\"2021-07-13T13:57:46.581Z\",\"archived\":false}]}"
+            "size": 207,
+            "text": "[\"H4sIAAAAAAAAAK2PO4/CMBCE/4tbQrR+YJxUUF5z1VUgipWzCCOHoLVzDeK/YyoQEo+CcmZnNPudBFMaY06iXZ9E6EQrtHK6seCsqAT1GGLxeDhg7GrkgGmiF344ZEafB6734zFkKneq/dCXyjZwyr/YU6kVGfFejYn4p4yYxjoDal4Jz4SZumUuAQVKT0FNofmTrp3NWwm1MmZ1LR67T2LIfhf+qSxsMSY6VzcmA1I5+YxJfosJ7Esme33WgX7DdB97ZNqcL2NB5s21AQAA\"]"
```
[the remark this answers is not in the document]
Comment (VDubber) [reply src/steps/owners/__recordings__/fetchOwners_1399607903/recording.har:231]: I vaguely remember running into this before. In the recording.ts file, try adding `mutations.unzipGzippedRecordingEntry(entry);` 
The most recent version of the testing sdk should provide the decoding.

File: src/client.ts [hunk 2]
```diff
@@ -4,49 +4,78 @@ import {
   IntegrationProviderAuthenticationError,
 } from '@jupiterone/integration-sdk-core';
 import * as hubspot from '@hubspot/api-client';
+import {
+  PublicOwner,
+  CollectionResponsePublicOwnerForwardPaging,
+} from '@hubspot/api-client/lib/codegen/crm/owners';
```
[the remark this answers is not in the document]
Comment (RonaldEAM) [reply src/client.ts:10]: Yes they don't export these in the index, but I'll check if I can find another way

File: src/types.ts [hunk 3]
```diff
@@ -1,53 +1,24 @@
...
-export interface Paging {
-  next: Next;
-}
-
-export interface Next {
-  after: string;
-  link: string;
-}
-
-export interface Owner {
```
[the remark this answers is not in the document]
Comment (RonaldEAM) [reply src/types.ts:36]: Yes

File: src/client.ts [hunk 4]
```diff
@@ -4,49 +4,78 @@ import {
...
     }
   }
 
   public async verifyAuthentication(): Promise<void> {
     try {
-      const tokens = await this.hubspotClient.crm.owners.defaultApi.getPage();
-      if (!tokens.body) {
-        throw new Error('Provider authentication failed');
-      }
+      await this.hubspotClient.crm.owners.ownersApi.getPage();
```
[the remark this answers is not in the document]
Comment (RonaldEAM) [reply src/client.ts:74]: Yes it does

File: src/types.ts [hunk 5]
```diff
@@ -1,3 +1,13 @@
+import { Client } from '@hubspot/api-client';
+
+type Awaited<T> = T extends Promise<infer ReturnType> ? ReturnType : T;
+
+export type OwnersResponse = Awaited<
```
Comment (VDubber) [inline src/types.ts:5]: Cool!

**Review by VDubber — APPROVED** [review]
Looks great! 
Consider releasing to a single account if you are unsure how the pagination will respond in customer scenarios.

**Pull request merged**